utils/notification_service.py

```diff
@@ -38,183 +64,607 @@ def handle_test_results(test_results):
-                "pipeline": "run_all_tests_tf_multi_gpu_test_reports/tests_tf_pipeline_multi_gpu_[].txt",
-            },
-            "Torch Multi GPU": {
-                "common": "run_all_tests_torch_multi_gpu_test_reports/tests_torch_multi_gpu_[].txt",
-                "pipeline": "run_all_tests_torch_multi_gpu_test_reports/tests_torch_pipeline_multi_gpu_[].txt",
-            },
-            "Torch Cuda Extensions Single GPU": {
-                "common": "run_tests_torch_cuda_extensions_gpu_test_reports/tests_torch_cuda_extensions_gpu_[].txt"
-            },
-            "Torch Cuda Extensions Multi GPU": {
-                "common": "run_tests_torch_cuda_extensions_multi_gpu_test_reports/tests_torch_cuda_extensions_multi_gpu_[].txt"
+                individual_reports.append(report)
+
+        category_failures_report = "\n".join(sorted(individual_reports))
+
+        return {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f"The following modeling categories had failures [single|multi]:\n{category_failures_report}",
             },
         }
-    else:
-        file_paths = {
-            "TF Single GPU": {"common": "run_all_tests_tf_gpu_test_reports/tests_tf_gpu_[].txt"},
-            "Torch Single GPU": {"common": "run_all_tests_torch_gpu_test_reports/tests_torch_gpu_[].txt"},
-            "TF Multi GPU": {"common": "run_all_tests_tf_multi_gpu_test_reports/tests_tf_multi_gpu_[].txt"},
-            "Torch Multi GPU": {"common": "run_all_tests_torch_multi_gpu_test_reports/tests_torch_multi_gpu_[].txt"},
-            "Torch Cuda Extensions Single GPU": {
-                "common": "run_tests_torch_cuda_extensions_gpu_test_reports/tests_torch_cuda_extensions_gpu_[].txt"
+
+    @property
+    def model_failures(self) -> Dict:
+        # Obtain per-model failures
+        def per_model_sum(model_category_dict):
+            return dicts_to_sum(model_category_dict["failed"].values())
+
+        failures = {k: per_model_sum(v) for k, v in self.model_results.items() if sum(per_model_sum(v).values())}
+
+        model_reports = []
+        other_module_reports = []
+        for key, value in failures.items():
+            device_report = self.get_device_report(value, rjust=3)
+
+            if sum(value.values()):
+                report = f"{key}"
+                if device_report:
+                    report = f"[{device_report}] {report}"
+
+                if key in NON_MODEL_TEST_MODULES:
+                    other_module_reports.append(report)
+                else:
+                    model_reports.append(report)
+
+        model_failures_report = "\n".join(sorted(model_reports, key=lambda s: s.split("] ")[-1]))
+        module_failures_report = "\n".join(sorted(other_module_reports, key=lambda s: s.split("] ")[-1]))
+
+        return {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
```

Comment:
```suggestion
                "type": "markdown",
```
or is the typo intentional?

Reply:
Yes, that's what the Slack sdk expects